### PR TITLE
Added "Introduction to C" as a new tutorial under Tech Stacks

### PR DESCRIPTION
**Pull request by SleepyyPlayz**
This is a culmination of my low level programming knowledge and a few topics most commonly used in C programming. It is by no means complete, since there are still topics that are not in here (e.g. function pointers, file interactions, taking parameters from the command line through the main function, etc.), but the ones that are in here are detailed enough to lay a solid foundation for understanding low level programming concepts, and to build skill thereupon.

- Chris Wangzheng Jiang

### CODE REVIEW

File: Topics/Tech_Stacks/C/Memory_Management.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
+# Memory Management
```
[the remark this answers is not in the document]
Comment (VictorYao17) [reply Topics/Tech_Stacks/C/Memory_Management.md:1]: I agree, I think adding a section explaining what the heap and stack are and why the heap is used would be helpful for beginners.

File: Topics/Tech_Stacks/C/Arrays.md
[the remark this answers is not in the document]
Comment (SleepyyPlayz) [reply Topics/Tech_Stacks/C/Arrays.md:1]: Added `sizeof()` explanation into Memory Management.

[the remark this answers is not in the document]
Comment (VictorYao17) [reply Topics/Tech_Stacks/C/Arrays.md:1]: Looks good

**Comment by SleepyyPlayz** [comment]
All the comments of this PR are resolved. However, since I don't have write access directly to the repo, I'll most likely have one of my team members merge this PR on my behalf.
(message for the TA)

**Pull request merged**